src/Evaluator.hs

```diff
@@ -24,7 +24,7 @@ evalExpr (Mul x y) = (*) <$> evalExpr x <*> evalExpr y
 evalExpr (Div x y) = do
     divisor <- evalExpr y
     if divisor == 0
-        then return (1 / 0) -- Return Infinity or some other default value
+        then return (1 / 0)
```

Comment:
Don't return infinities or any default values. You know techniques to express failure in a functional language.